### PR TITLE
Tweaks to the AI core construction

### LINKED ISSUE #16172: The R glass in AI construction is still acts like it's in your hand after application.
When you apply the R glass to a new AI core you act like you still have the glass in hand. The sprite vanishes but say you try and pick up something else like your tools and it's like you still have the glass in hand. You can drop whatever is in hand to rid the invisible thing. I had only 2 sheets of R glass in hand before using it on the AI core.
This has happened several times in the past and I assumed it was just due to stacks being bugged but it's still an issue so it's likely a bug in the core construction.

### PR DESCRIPTION
**Pull request by DamianX**
Fixes #16172 among unreported others

### CODE REVIEW

File: code/game/machinery/computer/ai_core.dm [hunk 1]
```diff
@@ -30,48 +43,43 @@
...
 		if(SECURED_CIRCUITBOARD)
 			if(isscrewdriver(P) && circuit)
 				playsound(loc, 'sound/items/Screwdriver.ogg', 50, 1)
 				to_chat(user, "<span class='notice'>You unfasten the circuit board.</span>")
 				state = UNSECURED_CIRCUITBOARD
-				icon_state = "1"
-			if(istype(P, /obj/item/stack/cable_coil))
-				if(P:amount >= 5)
+			var/obj/item/stack/cable_coil/cable_coil = P
+			if(istype(cable_coil))
```
Comment (SonixApache) [inline code/game/machinery/computer/ai_core.dm]: if(iscoil(P))

Comment (Shadowmech88) [reply code/game/machinery/computer/ai_core.dm]: Not that you're incorrect, but `P` needs to be cast as `/obj/item/stack/cable_coil` anyway in order to access its `amount` var.

**Comment by PJB3005** [comment]
Would it be possible to move the entire thing to use construction datums?

**Comment by DamianX** [comment]
I've never worked with construction datums, but I guess one day maybe.

**Pull request merged**